### PR TITLE
Add linting checks for PRs via Travis CI

### PR DESCRIPTION
**Pull request by andrewbonney**
Work in progress, subject to confirmation that was can use a suitable Travis account.

### CODE REVIEW

File: IS0402Test.py [hunk 1]
```diff
@@ -556,7 +556,8 @@ def test_21_2(self, test):
         description = "test_21_2"
 
         # Initial test cases based on the examples in NMOS documentation
-        # See https://github.com/AMWA-TV/nmos-discovery-registration/blob/v1.2.x/docs/2.5.%20APIs%20-%20Query%20Parameters.md#pagination
+        # See https://github.com/AMWA-TV/nmos-discovery-registration/blob/v1.2.x/docs/2.5.%20APIs%20-%20Query%20
+        # Parameters.md#pagination
```
Comment (garethsb-sony) [inline IS0402Test.py]: Please, no.

File: IS0402Test.py [hunk 2]
```diff
@@ -843,7 +847,7 @@ def test_21_7(self, test):
                                   expected_ids=ids,
                                   expected_since="0:0", expected_until=ts[-1], expected_limit=count)
 
-        resources = response[1].json() # valid json guaranteed by check_paged_response
+        resources = response[1].json()  # valid json guaranteed by check_paged_response
```
Comment (garethsb-sony) [inline IS0402Test.py]: Why's that space necessary?

File: IS0402Test.py (hunk 1 above)
Comment (andrewbonney) [reply IS0402Test.py]: I'm not sure what to do as an alternative without breaching the line length...

File: IS0402Test.py (hunk 2 above)
Comment (andrewbonney) [reply IS0402Test.py]: It's enforcing the rule from https://www.python.org/dev/peps/pep-0008/#inline-comments, second sentence.

File: IS0402Test.py [hunk 3]
```diff
@@ -1334,7 +1338,8 @@ def test_24_1(self, test):
             elif r.status_code == 200 and len(r.json()) > 0 or r.status_code != 200:
                 return test.OPTIONAL("Query API signalled that it does not support RQL queries. This may be "
                                      "important for scalability.",
-                                     "https://github.com/AMWA-TV/nmos/wiki/IS-04#registries-resource-query-language-rql")
+                                     "https://github.com/AMWA-TV/nmos/wiki/IS-04#registries-"
+                                     "resource-query-language-rql")
```
Comment (garethsb-sony) [inline IS0402Test.py]: In terms of being able to search for a URL of interest in the code-base, this is another place that breaking somewhere else might be better, e.g. before the fragment '#', if we are forced to break at all.

Or I had been wondering about introducing constants like `NMOS_WIKI_URL = "https://github.com/AMWA-TV/nmos/wiki"` (and similarly for the amwa-tv.github.io links) so we could write `NMOS_WIKI_URL + "/IS-04#registries-resource-query-language-rql"`?

File: IS0501Test.py [hunk 4]
```diff
@@ -55,8 +55,8 @@ def test_01(self, test):
         dest = ""
         valid, result = self.is05_utils.checkCleanRequestJSON("GET", dest)
         if valid:
-            msg = "Got the wrong json from {} - got {}. Please check json matches the spec, including trailing slashes" \
-                .format(dest, result)
+            msg = "Got the wrong json from {} - got {}. Please check json matches the spec, including trailing " \
+                  "slashes".format(dest, result)
```
Comment (garethsb-sony) [inline IS0501Test.py]: Is the line-continuation '\' needed in this and the next one? I haven't got my head around why it sometimes is, and why it sometimes isn't.

File: IS05Utils.py [hunk 5]
```diff
@@ -604,7 +604,7 @@ def park_resource(self, resource_type, resource_id):
         valid, response = self.checkCleanRequestJSON("PATCH", url, data=data)
         if valid:
             try:
-                staged_params = response['transport_params']
+                response.get('transport_params')
```
Comment (garethsb-sony) [inline IS05Utils.py]: Personally I think this is over-aggressive linting that actually reduces the code readability. :-(

File: IS0501Test.py (hunk 4 above)
Comment (andrewbonney) [reply IS0501Test.py]: If the statement is fully enclosed by brackets (such as an argument to a method) you don't need the continuation '\'. In this case it either requires the '\', or enclosing the string in brackets.

File: IS05Utils.py (hunk 5 above)
Comment (andrewbonney) [reply IS05Utils.py]: We can disable some linting if needed (probably in individual cases too), but there are lots of cases where an unused variable is an indication of a bug.

File: IS0501Test.py (hunk 4 above)
Comment (garethsb-sony) [reply IS0501Test.py]: Oh! Thanks. Makes sense.

File: IS05Utils.py (hunk 5 above)
Comment (garethsb-sony) [reply IS05Utils.py]: True. Sigh.

File: IS0402Test.py (hunk 2 above)
Comment (garethsb-sony) [reply IS0402Test.py]: OK. Thanks. (I probably should have just put that comment on the previous line.)

File: IS0402Test.py (hunk 1 above)
Comment (garethsb-sony) [reply IS0402Test.py]: Looks like we either add `# noqa: E501` (or whatever code it is) at the end of the line (see [in-line-ignoring-errors](http://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors)), or put the word `See` on its own line or omit it entirely (see [max-line-length](http://flake8.pycqa.org/en/3.1.1/user/options.html?highlight=line%20length#cmdoption-flake8--max-line-length)).

**Comment by andrewbonney** [comment]
@garethsb-sony I'm not planning to deal with this until Travis is set up (as I want something to prove it can fail), but there are a few unused 'description' variables in the pagination tests. I just wanted to check before resolving them whether that's intentional or not?

**Comment by garethsb-sony** [comment]
Fixed.

**Comment by garethsb-sony** [comment]
Oh, oops... just read that again... you wanted some outstanding lint... D'oh.

**Comment by garethsb-sony** [comment]
One unused description remaining, in `test_21_8`. 😢😂

**Comment by andrewbonney** [comment]
Ha, thanks!

**Pull request merged**